code/modules/mob/living/basic/minebots/minebot_ai.dm

```diff
@@ -0,0 +1,306 @@
+/datum/ai_controller/basic_controller/minebot
+	blackboard = list(
+		BB_TARGETING_STRATEGY = /datum/targeting_strategy/basic/always_check_factions,
+		BB_PET_TARGETING_STRATEGY = /datum/targeting_strategy/basic/not_friends,
+		BB_MINIMUM_SHOOTING_DISTANCE = 3,
+		BB_MINEBOT_PLANT_MINES = TRUE,
+		BB_MINEBOT_REPAIR_DRONE = TRUE,
+		BB_MINEBOT_AUTO_DEFEND = TRUE,
+		BB_BLACKLIST_MINERAL_TURFS = list(),
+		BB_AUTOMATED_MINING = FALSE,
+		BB_OWNER_SELF_HARM_RESPONSES = list(
+			"Please stop hurting yourself.",
+			"There is no need to do that.",
+			"Your actions are illogical.",
+			"Please make better choices.",
+			"Remember, you have beaten your worst days before.",
+		),
+		BB_OWNER_FRIENDLY_FIRE_APOLOGIES = list(
+			"Sorry.",
+			"My fault.",
+			"Oops.",
+		),
+	)
+
+	ai_movement = /datum/ai_movement/jps
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/pet_planning,
+		/datum/ai_planning_subtree/befriend_miners,
+		/datum/ai_planning_subtree/minebot_maintain_distance,
+		/datum/ai_planning_subtree/basic_ranged_attack_subtree/minebot,
+		/datum/ai_planning_subtree/find_and_hunt_target/hunt_ores/minebot,
+		/datum/ai_planning_subtree/minebot_mining,
+		/datum/ai_planning_subtree/flee_target/minebot_gibtonite,
+		/datum/ai_planning_subtree/locate_dead_humans,
+	)
+	ai_traits = AI_FLAG_PAUSE_DURING_DO_AFTER
+
+/datum/ai_planning_subtree/befriend_miners/select_behaviors(datum/ai_controller/controller, seconds_per_tick)
+	if(!controller.blackboard_key_exists(BB_MINER_FRIEND))
+		return
+	controller.queue_behavior(/datum/ai_behavior/befriend_target, BB_MINER_FRIEND)
+
+/// Find dead humans and report their location on the radio
+/datum/ai_planning_subtree/locate_dead_humans/select_behaviors(datum/ai_controller/controller, seconds_per_tick)
+	if(controller.blackboard_key_exists(BB_NEARBY_DEAD_MINER))
+		controller.queue_behavior(/datum/ai_behavior/send_sos_message, BB_NEARBY_DEAD_MINER)
+		return SUBTREE_RETURN_FINISH_PLANNING
+	controller.queue_behavior(/datum/ai_behavior/find_and_set/unconscious_human, BB_NEARBY_DEAD_MINER, /mob/living/carbon/human)
+
+/datum/ai_behavior/find_and_set/unconscious_human/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	// For now, we only look for friends, and we only befriend the person
+	// who created us, so we're not screaming on comms about every single
+	// dead player that we find
+	if(!controller.blackboard_key_exists(BB_FRIENDS_LIST))
+		return
```

Comment:
return null here too? Pick one for consistency in this function imo

Reply:
Done.